cumulus/steps/ec2/alb.py

```diff
@@ -0,0 +1,135 @@
+from cumulus.chain import step
+from troposphere import (
+    Ref, Not, Equals, Join, ec2,
+    If, Output
+)
+from troposphere import elasticloadbalancingv2 as alb
+
+CLUSTER_SG_NAME = "%sSG"
+ALB_SG_NAME = "%sAlbSG"
+ALB_NAME = "%sLoadBalancer"
+TARGET_GROUP_DEFAULT = "DefaultTargetGroup%s"
+
+
+class Alb(step.Step):
+
+    def __init__(self,
+                 name
+                 ):
+        step.Step.__init__(self)
+
+        self.name = name
+
+    def handle(self, chain_context):
+        self.create_conditions(chain_context.template)
+        self.create_security_groups(chain_context.template)
+        self.create_default_target_group(chain_context.template)
+        self.create_load_balancer_alb(chain_context.template)
+        self.add_listener(chain_context.template)
+
+    def create_conditions(self, template):
+        template.add_condition(
+            "UseSSL",
+            Not(Equals(Ref("ALBCertName"), "")))
+        template.add_condition(
+            "UseIAMCert",
+            Not(Equals(Ref("ALBCertType"), "acm")))
+
+    def create_security_groups(self, template):
+        asg_sg = CLUSTER_SG_NAME % self.name
+        alb_sg = ALB_SG_NAME % self.name
+
+        # TODO: this SG shouldn't be required.. test.
+        template.add_resource(ec2.SecurityGroup(
```

Comment:
TODO!

Reply:
Should probably get to that eventually in the infra blueprint that this came from.

Reply:
we're not using the one where it came from anymore.

Reply:
I have removed this one, and I don't see any ramifications in the net result, so I will leave it out.